string player
bool node_initialized